components/header/index.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import './index.scss'
+import Logo from '../logo'
+import Header from './Header'
+import HeaderItem from './HeaderItem'
+import Avatar from '../avatar'
+import SearchBar from '../searchbar'
```

Comment:
This is actually why I asked you to send an early work. You have too many dependencies here. Header should not contain everything. There is a business logic and the view rendering in the same time. Try to do something like following:

```jsx
const AppBar = ({ children, className, tag: Tag = 'div', ...restProps}) => (
  <Tag 
    className={combine(modulize('app-bar', classNames), className)}
    {...restProps}
  >
    {children}
  </Tag>
)

const Brand = ({ ... }) => ...
AppBar.Brand = Brand

export ...

...
...


const DashboardAppBar = ({ user }) => (
  <AppBar className="dashboard-app-bar">
    <AppBar.Brand>Dashboard</AppBar.Brand>
    ...
    <AppBar.Avatar src={user.avatarUrl} alt={`${user.name}'s avatar`} />
  </AppBar>
)
```

The first part contains pure components that are as simple as possible and may freely go to the library. The last one is dashboard-dependent and stays in the dashboard.

Reply:
Well, my initial idea was to move `Header`, `HeaderItem`, `Avatar`, `SearchBar` to the library once it's done and in `components/header/` would stay only `index.js`. I guess it's mostly the same what you are proposing.

> ```js
> const Brand = ({ ... }) => ...
> AppBar.Brand = Brand
> ```

I do like to have it like that. I wanted to have it like `import {Header, Brand, HeaderItem} from '@oacore/design'`. Your suggestion seems much better. Thanks.


What I do not understand are functions `combine` & `modulize`. Where did you get them from?

Reply:
`combine` & `modulize` are abstract for this moment. I wanted to add an implementation as utilities into the library.

The first one may be [@JedWatson/classNames](https://github.com/JedWatson/classnames) but I do not really like that its name does not say what it does.

The second one is a simple map over an object.